src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/EventListener.java

```diff
@@ -164,25 +166,45 @@ protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent
      * @param event   the event.
      * @param project the project to build.
      */
-    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, AbstractProject project) {
+    protected void schedule(GerritTrigger t, GerritCause cause, GerritTriggeredEvent event, Job project) {
         BadgeAction badgeAction = new BadgeAction(event);
         //during low traffic we still don't want to spam Gerrit, 3 is a nice number, isn't it?
         int projectbuildDelay = t.getBuildScheduleDelay();
         if (cause instanceof GerritUserCause) {
             // it's a manual trigger, no need for a quiet period
             projectbuildDelay = 0;
-        } else if (project.getHasCustomQuietPeriod()
-                && project.getQuietPeriod() > projectbuildDelay) {
-            projectbuildDelay = project.getQuietPeriod();
+        } else if (project instanceof AbstractProject) {
+            // TODO: How about using ParameterizedJobMixIn.ParameterizedJob here?
```

Comment:
Yes, that is the usual interface for finding a `quietPeriod`.

I do not follow the logic relating to `hasCustomQuietPeriod`; this is definitely unusual. One of many “snowflake” features in this plugin…

More broadly, my impression is that quiet periods are of little importance to people not using archaic systems like CVS that lack atomic commits.